pulsar/producer_partition.go

```diff
@@ -397,19 +460,48 @@ func (p *partitionProducer) internalSend(request *sendRequest) {
 	msg := request.msg
 
 	payload := msg.Payload
+
 	var schemaPayload []byte
 	var err error
-	if p.options.Schema != nil {
-		schemaPayload, err = p.options.Schema.Encode(msg.Value)
+	if msg.Value != nil && msg.Payload != nil {
+		p.log.Error("Can not set Value and Payload both")
+		return
+	}
+
+	if p.options.DisableMultiSchema {
+		if msg.Schema != p.options.Schema {
```

Comment:
I am not sure if we want to compare the pointer or the actual value. Shema is an interface. So the current evaluation just make sure they are pointing to the same reference. Do you want to compare the value of Schema?

Reply:
Yes. Maybe I can compare the value of Schema with this.
`msg.Schema != nil && p.options.Schema != nil && msg.Schema.GetSchemaInfo().hash() != p.options.Schema.GetSchemaInfo().hash()`